File: 솝트 알고리즘 스터디/줄서는 방법.js
function getFactorial(num) { // 팩토리얼 구하기
    let fact = 1;
    for (let i = 2; i <= num; i++) fact *= i;
    return fact;
}

function solution(n, k) {
    let div;
    let mod = k;
    let answer = [];
    let cand = [...Array(n).keys()].map(x => x + 1) // range()
    for (let i = n - 1; i > 0; i--) {
        div = parseInt(mod / getFactorial(i)) // 몫 구할 때
        mod = mod % getFactorial(i)
        if (mod === 0) { // 나머지 0일 때
            return answer.concat(cand.splice(div - 1, 1), cand.reverse())
        } // splice는 원래배열 변경, 배열 아닌 요소도 이어붙이고 싶을 때 concat
        answer.push(parseInt(cand.splice(div, 1)))
    }
    return [1]
}

File: 솝트 알고리즘 스터디/줄서는 방법.py
import math
def solution(n, k):
    mod = k
    answer, cand = [], list(range(1, n+1))
    for i in range(n-1, 0, -1):
        div, mod = divmod(mod, math.factorial(i))
        if mod == 0: # 나머지 0일 때
            answer += [cand.pop(div-1)] + cand[::-1]
            return answer
        answer.append(cand.pop(div))  # 나머지 0이 아닐 때
    return [1] # n이 1일 떄